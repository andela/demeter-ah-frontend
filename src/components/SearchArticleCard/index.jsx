import React from 'react';
import { Link } from 'react-router-dom';

const SearchArticleCard = ({
  image,
  category,
  authorname,
  title,
  slug,
  readTime,
  createdAt,
  tags
}) => {
  const setTags = tags && tags.length ? (
    tags.map(tag => (
      <div className="mr-2" key={tag.id}>
        <p className="sm:text-xs md:text-sm rounded-full sm:px-2 md:px-4 cursor-pointer border-2
          border-solid border-purple-200 text-purple-200 bg-white"
        >
          {tag.name}
        </p>
      </div>
    ))
  ) : (
    <p className="text-xs" />
  );

  return (
    <>
      <div className="md:flex md:flex-row flex-row lg:min-w-96 m-auto text-left justify-start pl-4 pt-4 pb-8
        bg-white rounded-sm border-b-2 border-gray-30 mb-0 md:min-w-84"
      >
        <div className="m-4 ml-0 mb-0 w-full h-48 md:w-48 md:h-24 rounded shadow-lg">
          <img src={image || '/placeholder.png'} alt="Article" width="100%" height="100%" className="h-full object-cover" />
        </div>
        <div className="ml-0 flex-grow-0 justify-between flex flex-col max-h-48">
          <div className="flex flex-col">
            <Link to={`/articles/${slug}`}>
              <h2 className="text-1.5xl lg:text-xl capitalize">{title || 'Title of this amazing article'}</h2>
            </Link>
            <h4 className="text-base m-0">
              {'by '}
              <span className="font-semibold">{ authorname || 'Paul Mark'}</span>
            </h4>
          </div>
          <div className="flex justify-between lg:min-w-118 md:min-w-96">
            <div className="flex flex-wrap text-xs">
              <p className="pr-4 mt-1">{category || 'Others'}</p>
              <p className="pr-4 mt-1">{readTime || '6 min'}</p>
              <p className="font-semibold mt-1 pr-4">{createdAt || 'July 12 2019'}</p>
              <div className="flex flex-row mt-2 md:mt-0 ">
                {setTags}
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default SearchArticleCard;
