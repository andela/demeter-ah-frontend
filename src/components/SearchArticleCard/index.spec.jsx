import React from 'react';
import { shallow } from 'enzyme';
import SearchArticleCard from './index';

describe('SearchArticleCard Component', () => {
  const tags = [
    {
      id: 3,
      name: 'react'
    }
  ];
  it('Should render without errors', () => {
    const component = shallow(<SearchArticleCard tags={tags} />);
    expect(component).toMatchSnapshot();
  });

  it('Should render without errors when no tags exist', () => {
    const component = shallow(<SearchArticleCard />);
    expect(component).toMatchSnapshot();
  });
});
