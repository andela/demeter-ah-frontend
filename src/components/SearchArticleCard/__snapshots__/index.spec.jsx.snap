// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchArticleCard Component Should render without errors 1`] = `
<Fragment>
  <div
    className="md:flex md:flex-row flex-row lg:min-w-96 m-auto text-left justify-start pl-4 pt-4 pb-8 bg-white rounded-sm border-b-2 border-gray-30 mb-0 md:min-w-84"
  >
    <div
      className="m-4 ml-0 mb-0 w-full md:w-48 h-24 rounded shadow-lg"
    >
      <img
        alt="Article"
        className="h-full"
        height="100%"
        src="/article/no-image.png"
        width="100%"
      />
    </div>
    <div
      className="ml-0 flex-grow-0 justify-between flex flex-col max-h-48"
    >
      <div
        className="flex flex-col"
      >
        <Link
          to="/articles/undefined"
        >
          <h2
            className="text-1.5xl lg:text-xl capitalize"
          >
            Title of this amazing article
          </h2>
        </Link>
        <h4
          className="text-base m-0"
        >
          by 
          <span
            className="font-semibold"
          >
            Paul Mark
          </span>
        </h4>
      </div>
      <div
        className="flex justify-between lg:min-w-118 md:min-w-96"
      >
        <div
          className="flex flex-wrap text-xs"
        >
          <p
            className="pr-4 mt-1"
          >
            Others
          </p>
          <p
            className="pr-4 mt-1"
          >
            6 min
          </p>
          <p
            className="font-semibold mt-1 pr-4"
          >
            July 12 2019
          </p>
          <div
            className="flex flex-row mt-2 md:mt-0 "
          >
            <div
              className="mr-2"
              key="3"
            >
              <p
                className="sm:text-xs md:text-sm rounded-full sm:px-2 md:px-4 cursor-pointer border-2 border-solid border-purple-200 text-purple-200 bg-white"
              >
                react
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`SearchArticleCard Component Should render without errors when no tags exist 1`] = `
<Fragment>
  <div
    className="md:flex md:flex-row flex-row lg:min-w-96 m-auto text-left justify-start pl-4 pt-4 pb-8 bg-white rounded-sm border-b-2 border-gray-30 mb-0 md:min-w-84"
  >
    <div
      className="m-4 ml-0 mb-0 w-full md:w-48 h-24 rounded shadow-lg"
    >
      <img
        alt="Article"
        className="h-full"
        height="100%"
        src="/article/no-image.png"
        width="100%"
      />
    </div>
    <div
      className="ml-0 flex-grow-0 justify-between flex flex-col max-h-48"
    >
      <div
        className="flex flex-col"
      >
        <Link
          to="/articles/undefined"
        >
          <h2
            className="text-1.5xl lg:text-xl capitalize"
          >
            Title of this amazing article
          </h2>
        </Link>
        <h4
          className="text-base m-0"
        >
          by 
          <span
            className="font-semibold"
          >
            Paul Mark
          </span>
        </h4>
      </div>
      <div
        className="flex justify-between lg:min-w-118 md:min-w-96"
      >
        <div
          className="flex flex-wrap text-xs"
        >
          <p
            className="pr-4 mt-1"
          >
            Others
          </p>
          <p
            className="pr-4 mt-1"
          >
            6 min
          </p>
          <p
            className="font-semibold mt-1 pr-4"
          >
            July 12 2019
          </p>
          <div
            className="flex flex-row mt-2 md:mt-0 "
          >
            <p
              className="text-xs"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
